src/test/unit_tests/test_nc_iam_basic_integration.js

```diff
@@ -0,0 +1,234 @@
+        });
+
+        mocha.it('list access keys - should have 1 access key', async function() {
+            const input = {
+                UserName: username2
+            };
+            const command = new ListAccessKeysCommand(input);
+            const response = await iam_account.send(command);
+            assert.equal(response.$metadata.httpStatusCode, 200);
+            assert.equal(response.AccessKeyMetadata.length, 1);
+            assert.equal(response.AccessKeyMetadata[0].UserName, username2);
+            assert.equal(response.AccessKeyMetadata[0].AccessKeyId, access_key_id);
+            assert.equal(response.AccessKeyMetadata[0].Status, ACCESS_KEY_STATUS_ENUM.ACTIVE);
+        });
+
+        mocha.it('get access key (last used)', async function() {
+            const input = {
+                AccessKeyId: access_key_id
+            };
+            const command = new GetAccessKeyLastUsedCommand(input);
+            const response = await iam_account.send(command);
+            assert.equal(response.$metadata.httpStatusCode, 200);
+            assert.equal(response.UserName, username2);
+            assert(response.AccessKeyLastUsed.LastUsedDate !== undefined);
+            assert(response.AccessKeyLastUsed.ServiceName !== undefined);
+            assert(response.AccessKeyLastUsed.Region !== undefined);
+        });
+
+        mocha.it('update access keys (active to inactive)', async function() {
+            const input = {
+                UserName: username2,
+                AccessKeyId: access_key_id,
+                Status: ACCESS_KEY_STATUS_ENUM.INACTIVE
+            };
+            const command = new UpdateAccessKeyCommand(input);
+            const response = await iam_account.send(command);
+            assert.equal(response.$metadata.httpStatusCode, 200);
+        });
+
+        mocha.it('list access keys - should have 1 access key (with inactive status)', async function() {
+            const input = {
+                UserName: username2
+            };
+            const command = new ListAccessKeysCommand(input);
+            const response = await iam_account.send(command);
+            assert.equal(response.$metadata.httpStatusCode, 200);
+            assert.equal(response.AccessKeyMetadata.length, 1);
+            assert.equal(response.AccessKeyMetadata[0].UserName, username2);
+            assert.equal(response.AccessKeyMetadata[0].AccessKeyId, access_key_id);
+            assert.equal(response.AccessKeyMetadata[0].Status, ACCESS_KEY_STATUS_ENUM.INACTIVE);
+        });
+
+        mocha.it('delete access keys', async function() {
+            const input = {
+                UserName: username2,
+                AccessKeyId: access_key_id
+            };
+            const command = new DeleteAccessKeyCommand(input);
+            const response = await iam_account.send(command);
+            assert.equal(response.$metadata.httpStatusCode, 200);
```

Comment:
besides the response was successful, should you check that they really were deleted and that another usage with these keys will throw access denied?

Reply:
In the `test_accountspace_fs.test.js` I check that I can't find the path by the deleted access key.
I don't have full usage as the integration is IAM to the server and I don't have a full IAM_S3 integration as you asked before.